crates/core/tedge_watchdog/src/systemd_watchdog.rs

```diff
@@ -29,6 +29,37 @@ pub async fn start_watchdog(tedge_config_dir: PathBuf) -> Result<(), anyhow::Err
     // Send ready notification to systemd.
     notify_systemd(process::id(), "--ready")?;
 
+    // Send heart beat notifications to systemd, to notify about its own health status
+    start_watchdog_for_self().await;
+
+    // Monitor health of tedge services
+    start_watchdog_for_tedge_services(tedge_config_dir).await;
+    Ok(())
+}
+
+async fn start_watchdog_for_self() {
+    tokio::spawn(async move {
+        match get_watchdog_sec("/lib/systemd/system/tedge-watchdog.service") {
+            Ok(interval) => loop {
+                let _ = notify_systemd(process::id(), "WATCHDOG=1").map_err(|e| {
+                    eprintln!("{}", e);
+                });
+                tokio::time::sleep(tokio::time::Duration::from_secs(interval / 4)).await;
+            },
+
+            Err(_) => {
+                warn!(
+                    "Watchdog is not enabled for tedge-watchdog: {}",
+                    WatchdogError::NoWatchdogSec {
+                        file: "tedge-watchdog.service".to_string()
+                    }
+                );
```

Comment:
Why ignore the error with `_` then to rebuild it as `WatchdogError::NoWatchdogSec { ... }`?